### PR TITLE
File view in channel (966)

### PR DESCRIPTION
**Pull request by huynguyennovem**
https://user-images.githubusercontent.com/29337364/157587172-db67cdb7-dbca-4aec-a60f-2d51fae71b28.mp4

### CODE REVIEW

File: lib/pages/chat/file_preview.dart [hunk 1]
```diff
@@ -69,7 +72,11 @@ class _FilePreviewState<T extends BaseChannelsCubit>
                 width: double.maxFinite,
                 height: double.maxFinite,
                 fit: BoxFit.contain,
-                imageUrl: file.downloadUrl,
```
Comment (EvGen94) [inline lib/pages/chat/file_preview.dart:72]: You don't have to do this, we've already changed the story, use downloadUrl, that's why it is here 
So we need the download icon

File: lib/pages/channel/channel_files/channel_files_widget.dart [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+              ),
+            ),
+          );
+        }
+        return Container(
+          margin: const EdgeInsets.all(16.0),
+          alignment: Alignment.center,
+          child: CircularProgressIndicator(
+            backgroundColor: const Color.fromRGBO(153, 162, 173, 0.4),
+            color: const Color(0xff004dff),
```
Comment (EvGen94) [inline lib/pages/channel/channel_files/channel_files_widget.dart:114]: Use theme colors - like surface 
We should not hardcoding it

File: lib/pages/channel/channel_files/channel_files_widget.dart [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+                itemCount: channels.length,
+                separatorBuilder: (BuildContext context, int index) => SizedBox(height: 12.0),
+                itemBuilder: (context, index) {
+                  return _buildChannelFileItem(channels[index]);
+                },
+              ),
+            ),
+          );
+        }
+        return Container(
```
Comment (EvGen94) [inline lib/pages/channel/channel_files/channel_files_widget.dart:109]: What if we don't have files in the channel - show the CircularProgressIndicator forever, this needs another solution for that case

File: lib/widgets/common/file_tile.dart [hunk 4]
```diff
@@ -110,7 +109,9 @@ class _FileTileState extends State<FileTile> {
         /// Read [1] for the detail
```
Comment (EvGen94) [inline lib/widgets/common/file_tile.dart:109]: /// Read [1] for the detail - outdated

File: lib/repositories/messages_repository.dart [hunk 5]
```diff
@@ -89,6 +103,9 @@ class MessagesRepository {
       queryParameters['page_token'] = afterMessageId;
       queryParameters['direction'] = 'future';
     }
+    if(withExistedFiles == true) {
+      queryParameters['filter'] = 'files';
+    }
     if (threadId == null) {
```
Comment (EvGen94) [inline lib/repositories/messages_repository.dart:109]: What about files in the threads, do we show them?

File: lib/pages/channel/channel_files/channel_files_widget.dart [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+          margin: const EdgeInsets.all(16.0),
+          alignment: Alignment.center,
+          child: CircularProgressIndicator(
```
Comment (EvGen94) [inline lib/pages/channel/channel_files/channel_files_widget.dart:112]: What if we don't have files in the channel - show the CircularProgressIndicator forever, this needs another solution for that case

**Comment by EvGen94** [comment]
What about viewing files in direct chats?

File: lib/pages/chat/file_preview.dart (hunk 1 above)
Comment (huynguyennovem) [reply lib/pages/chat/file_preview.dart:72]: I don't see the change for that in the mockup and description.

File: lib/pages/channel/channel_files/channel_files_widget.dart (hunk 2 above)
Comment (huynguyennovem) [reply lib/pages/channel/channel_files/channel_files_widget.dart:114]: right, I missed this

File: lib/widgets/common/file_tile.dart (hunk 4 above)
Comment (huynguyennovem) [reply lib/widgets/common/file_tile.dart:109]: Why is it outdated?

File: lib/repositories/messages_repository.dart (hunk 5 above)
Comment (huynguyennovem) [reply lib/repositories/messages_repository.dart:109]: Currently the API hasn’t  supported yet, let's wait for them

Comment (huynguyennovem) [reply lib/repositories/messages_repository.dart:109]: I'll ping Romaric about this

File: lib/pages/channel/channel_files/channel_files_widget.dart (hunk 6 above)
Comment (huynguyennovem) [reply lib/pages/channel/channel_files/channel_files_widget.dart:112]: hmm, we don't have design now. But I will add a message that notices "No files in channel". Is it ok?

**Comment by huynguyennovem** [comment]
> What about viewing files in direct chats?

We have no channel detail for direct chats now as I see, so no way to go to file view page now.

File: lib/pages/channel/channel_files/channel_files_widget.dart (hunk 6 above)
Comment (huynguyennovem) [reply lib/pages/channel/channel_files/channel_files_widget.dart:112]: Screenshot_1646969516

**Review by EvGen94 — APPROVED** [review]

**Pull request merged**